konduit-serving-pipeline/src/test/java/ai/konduitai/serving/pipeline/protocol/TestURL.java

```diff
@@ -0,0 +1,121 @@
+package ai.konduitai.serving.pipeline.protocol;
+
+import ai.konduit.serving.pipeline.api.protocol.FtpClient;
+import ai.konduit.serving.pipeline.api.protocol.URIResolver;
+import ai.konduit.serving.pipeline.api.protocol.handlers.KSStreamHandlerFactory;
+import ai.konduitai.serving.common.test.TestServer;
+import lombok.val;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.mockftpserver.fake.FakeFtpServer;
+import org.mockftpserver.fake.UserAccount;
+import org.mockftpserver.fake.filesystem.DirectoryEntry;
+import org.mockftpserver.fake.filesystem.FileEntry;
+import org.mockftpserver.fake.filesystem.WindowsFakeFileSystem;
+
+import java.io.*;
+import java.net.URL;
+import java.net.URLConnection;
+
+import static junit.framework.Assert.assertTrue;
+import static junit.framework.Assert.fail;
+
+public class TestURL {
+
+    private TestServer server;
+    private final static int PORT = 9090;
+    private final static String HOST = "localhost";
+    private static final String HTTP = "http://";
+    private static final String HTTPS = "http://";
+    private static final String FTP = "ftp://";
+
+    private String configData = StringUtils.EMPTY;
+
+    @Rule
+    public TemporaryFolder testDir = new TemporaryFolder();
+
+    @Before
+    public void setUp() throws Exception {
+        URL.setURLStreamHandlerFactory(new KSStreamHandlerFactory());
+        server = new TestServer(HTTP, HOST, PORT);
+        server.start();
+        configData = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("config/config.json"),
+                "UTF-8");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        server.stop();
+    }
+
+    @Test
+    public void testHttpURL() throws Exception {
+        String url = HTTP + HOST + ":" + PORT + "/src/test/resources/config/config.json";
+        String data = FileUtils.readFileToString(URIResolver.getFile(url), "UTF-8");
+        assertTrue(data.contains("pipelineSteps"));
+
+        url = HTTPS + HOST + ":" + PORT + "/src/test/resources/config/config.json";
+        data = FileUtils.readFileToString(URIResolver.getFile(url), "UTF-8");
+        assertTrue(data.contains("pipelineSteps"));
+    }
+
+    @Ignore("Windows bound manual test")
+    @Test
+    public void testFTP() throws IOException, InterruptedException {
+        FakeFtpServer fakeFtpServer = new FakeFtpServer();
```

Comment:
If this is windows-specific (that's not ideal though) we can use `assumeTrue(SystemUtils.IS_OS_WINDOWS)` (that's in commons lang 3) - it'll dynamically skip non-Windows.